### PR TITLE
chore: Refactor 'lib/templates/package.js' to use resolvers

### PR DESCRIPTION
**Pull request by mlakov**
# Refactor `lib/templates/package.js` to Use Resolvers Pattern

### Refactor

♻️ Refactors `lib/templates/package.js` to use an explicit `RESOLVERS` map for all field-level logic, consistent with how other templates in the project handle configuration resolution. Also extracts a shared `prune` utility into `lib/common/utils.js` and updates dependent files accordingly.

### Changes

* `lib/templates/package.js`: Introduces a frozen `RESOLVERS` object covering `title`, `vendor`, `version`, `partOfProducts`, `description`, `shortDescription`, and `ordId`. Replaces inline property construction and spread-override pattern in `createPackage` with resolver calls and `prune`. Removes the redundant second-pass `placeholders.replace` in `createPackages`. Exports `RESOLVERS` for external use and testing.

* `lib/common/utils.js`: Adds a generic `prune` utility function that removes `null`, `undefined`, empty arrays, and empty objects from a plain object. Exports it alongside existing utilities.

* `lib/ord.js`: Replaces the inline `prune` implementation (previously duplicated in this file) with the shared `utils.prune`, keeping the domain-specific unused-package filtering logic intact.

* `lib/templates/event-resource.js`: Imports and uses the shared `prune` utility. Moves the `exposedEntityTypes` field directly into the returned object (relying on `prune` to drop it when empty), removing the previous conditional spread.

* `__tests__/unit/common/utils.test.js`: Adds a full `describe("prune", ...)` suite covering strings, numbers, booleans, `null`, `undefined`, empty/non-empty arrays, and empty/non-empty objects.

* `__tests__/unit/templates/package.test.js`: Adds a comprehensive `describe("RESOLVERS", ...)` suite testing each resolver individually, including env overrides, fallback behaviour, placeholder substitution (`{namespace}`, `{type}`), and edge cases.

- [ ] 🔄 Regenerate and Update Summary

<details>
<summary>PR Bot Information</summary>

**Version:** `1.22.5`

- Output Template: [Default Template](https://github.tools.sap/intelligent-insights/i2-pull-request/blob/main/src/services/llm/prompts/summary_default_output_template.md)
- Event Trigger: `pull_request.opened`
- Correlation ID: `9ae14635-df64-42ad-a941-b8491c32e137`
- LLM: `anthropic--claude-4.6-sonnet`
- Summary Prompt: [Default Prompt](https://github.tools.sap/intelligent-insights/i2-pull-request/blob/main/src/services/llm/prompts/summary_instructions_prompt.md)
- File Content Strategy: Full file content
</details>

### CODE REVIEW

File: lib/templates/event-resource.js [hunk 1]
```diff
@@ -134,12 +133,11 @@ function createEventResourceTemplate(service, appConfig) {
         visibility: RESOLVERS.visibility(service, appConfig),
         partOfGroups: RESOLVERS.partOfGroups(service, appConfig),
         partOfPackage: RESOLVERS.partOfPackage(service, appConfig),
+        exposedEntityTypes: RESOLVERS.exposedEntityTypes(service, appConfig),
```
[the remark this answers is not in the document]
Comment (mlakov) [reply lib/templates/event-resource.js:136]: `prune` takes care of this.

File: lib/templates/package.js [hunk 2]
```diff
@@ -1,24 +1,58 @@
...
+    return prune({
+        title: RESOLVERS.title(appConfig),
+        vendor: RESOLVERS.vendor(appConfig),
+        version: RESOLVERS.version(appConfig),
+        partOfProducts: RESOLVERS.partOfProducts(appConfig, products),
+        description: RESOLVERS.description(appConfig, label, visibility),
+        ordId: RESOLVERS.ordId(appConfig, label, visibility, resourceType),
+        shortDescription: RESOLVERS.shortDescription(appConfig, label, visibility),
+
+        ..._.omit(appConfig?.env?.packages?.[0], Object.keys(RESOLVERS)),
```
[the remark this answers is not in the document]
Comment (mlakov) [reply lib/templates/package.js:54]: Not relevant at all.

**Review by zongqichen — APPROVED** [review]
LGTM
